From be58eed849f5457bb49b79e94aa6a26971ba6deb Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Fri, 15 Dec 2017 11:11:17 -0800
Subject: [PATCH] Ensure all fields in iscsiuio IPC response are set

Make sure all fields in the response strcuture are set,
or info from the stack can be leaked to our caller.
Found by Qualsys.
---
 iscsiuio/src/unix/iscsid_ipc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/iscsiuio/src/unix/iscsid_ipc.c b/iscsiuio/src/unix/iscsid_ipc.c
index b5d7051b0558..c3b10e03774a 100644
--- a/iscsiuio/src/unix/iscsid_ipc.c
+++ b/iscsiuio/src/unix/iscsid_ipc.c
@@ -955,6 +955,8 @@ int process_iscsid_broadcast(int s2)
 	LOG_DEBUG(PFX "recv iscsid request: cmd: %d, payload_len: %d",
 		  cmd, payload_len);
 
+	memset(&rsp, 0, sizeof(rsp));
+
 	switch (cmd) {
 	case ISCSID_UIP_IPC_GET_IFACE:
 		size = fread(&data->u.iface_rec, payload_len, 1, fd);
-- 
2.17.2

